src/main/java/Duke.java

```diff
@@ -1,10 +1,126 @@
+        System.out.println("\t Bye. Hope to see you again soon!\n");
+        System.out.println("____________________________________________________________\n");
+    }
+
+    public static void printSpecificTask(int index, Task[] commands, String message) {
+        if (!message.equals("")) {
+            System.out.println("\t " + message);
+        }
+        System.out.println("\t" + index + "." + commands[index]);
+    }
+
+    public static void main(String[] args) {
+        printWelcomeMessage();
+        Task[] commands = createList();
+        boolean not_finished = true;
+        while (not_finished == true) {
+            String line = ask();
+            if (line.equals("bye")) {
+                printByeMessage();
+                not_finished = false;
+            } else if (line.equals("list")) {
+                displayCommandsList(commands);
+            } else if (line.split(" ")[0].equals("mark")) {
+                int index = Integer.parseInt(line.split(" ")[1]);
+                commands[index].setDone(true);
+                printDashLine();
+                printSpecificTask(index, commands, "Nice! I've marked this task as done:");
+                printDashLine();
+            } else if (line.split(" ")[0].equals("unmark")) {
+                int index = Integer.parseInt(line.split(" ")[1]);
+                commands[index].setDone(false);
+                printDashLine();
+                printSpecificTask(index, commands, "OK, I've marked this task as not done yet:");
+                printDashLine();
+            } else if (line.split(" ")[0].equals("todo")) {
+                if (line.split(" ").length == 1) {
+                    printDashLine();
+                    System.out.println("\t OOPS!!! The description of a todo cannot be empty.");
+                    printDashLine();
+                } else {
+                    commands[nextIndexInList] = new Todo(line.substring(5));
+                    nextIndexInList++;
+                    printDashLine();
+                    printSpecificTask(nextIndexInList - 1, commands, "Got it. I've added this task:");
+                    printDashLine();
+                }
+
+            } else if (line.split(" ")[0].equals("event")) {
+                int indexFrom = line.indexOf("/");
+                int indexTo = line.indexOf("/", indexFrom + 1);
+                commands[nextIndexInList] = new Event(line.substring(6, indexFrom - 1),
+                        line.substring(indexFrom + 6, indexTo - 1), line.substring(indexTo + 4));
+                nextIndexInList++;
+                printDashLine();
+                printSpecificTask(nextIndexInList - 1, commands, "Got it. I've added this task:");
+                printDashLine();
+            } else if (line.split(" ")[0].equals("deadline")) {
+                int index_by = line.indexOf("/");
+                commands[nextIndexInList] = new Deadline(line.substring(9, index_by - 1), line.substring(index_by + 4));
```

Comment:
Maybe you can try another way of parsing strings to avoid these magic numbers here?